### DISCUSSION TITLE
wasm build `rego_type_error: undefined ref`

### DISCUSSION THREAD

**Discussion by shm12** [open-policy-agent/community#305] [OPA and Rego] [answered]
Hi all. I'm facing strange behavior.
I have an `example.rego` file:
```rego
package play

allow {
	b := foo("bar")
	b == true
}

bar(val) = value {
	value = true
}

foo(val) = value {
	value = bar(val)
}
```
Now, if I run it, or build it - `opa build example.rego` , it's alright. But if I try to build a WASM bundle, I get:
```
$ opa build -t wasm -e play example.rego
error: 1 error occurred: example.rego:1: rego_type_error: undefined ref: data.play.foo
        data.play.foo
        ^^^^^^^^^^^^^
        have: (any) => boolean
```
Is it a bug, or is something wrong with the Rego code?

If It's only one function, like:
```rego
package play

allow {
	b := bar("bar")
	b == true
}

bar(val) = value {
	value = true
}
```
I don't get this error. Only where a function calls another function.

**Comment by srenatus** [comment, marked as answer]
Can you try the command line argument mentioned in https://github.com/open-policy-agent/opa/issues/4411?

**Reply by shm12** [reply]
It worked :)

I didn't quite understand why though.
Putting here the command for people who get to here:
```
opa build -t wasm -e play example.rego --prune-unused
```
